app/publish/http.go

```diff
@@ -97,3 +105,40 @@ func retryPolicy(ctx context.Context, resp *http.Response, err error) (bool, err
 
 	return false, nil
 }
+
+// parseRequestIP returns the request IP address.
+func parseRequestIP(headers http.Header, remoteAddr string) string {
```

Comment:
Is there any way we can re-use existing mechanisms for IP detection and user auth?

Reply:
updated to use the implementation from `ip` package